testing_utils/check_covidcast_port.py

```diff
@@ -0,0 +1,169 @@
+      COVIDcast signal documentation. Defaults to ``"county"``.
+    geo_values: The geographies to fetch data for. The default, ``"*"``,
+      fetches all geographies. To fetch one geography, specify its ID as a
+      string; multiple geographies can be provided as an iterable (list, tuple,
+      ...) of strings.
+    as_of: Fetch only data that was available on or before this date,
+      provided as a ``datetime.date`` object. If ``None``, the default, return
+      the most recent available data. If ``time_type == "week"``, then
+      this is rounded to the epiweek containing the day (i.e. the previous Sunday).
+    lag: Integer. If, for example, ``lag=3``, fetch only data that was
+      published or updated exactly 3 days after the date. For example, a row
+      with ``time_value`` of June 3 will only be included in the results if its
+      data was issued or updated on June 6. If ``None``, the default, return the
+      most recently issued data regardless of its lag.
+    time_type: The temporal resolution to request this data. Most signals
+      are available at the "day" resolution (the default); some are only
+      available at the "week" resolution, representing an MMWR week ("epiweek").
+    :returns: A Pandas data frame with matching data, or ``None`` if no data is
+      returned. Each row is one observation on one day in one geographic location.
+      Contains the following columns:
+    """
+    if start_day > end_day:
+        raise ValueError(
+            "end_day must be on or after start_day, but " f"start_day = '{start_day}', end_day = '{end_day}'"
+        )
+
+    response = Epidata.covidcast(
+        data_source,
+        signal,
+        time_type=time_type,
+        geo_type=geo_type,
+        time_values=Epidata.range(start_day.strftime("%Y%m%d"), end_day.strftime("%Y%m%d")),
+        geo_value=geo_values,
+        as_of=as_of,
+        lag=lag,
+    )
+    if response["result"] != 1:
+        # Something failed in the API and we did not get real metadata
+        raise RuntimeError("Error when fetching signal data from the API", response["message"])
+
+    api_df = pd.DataFrame.from_dict(response["epidata"])
+    api_df["issue"] = pd.to_datetime(api_df["issue"], format="%Y%m%d")
+    api_df["time_value"] = pd.to_datetime(api_df["time_value"], format="%Y%m%d")
+    api_df.drop("direction", axis=1, inplace=True)
+    api_df["data_source"] = data_source
+    api_df["signal"] = signal
+
+    return api_df
+
+def check_metadata():
+    expected_df = covidcast.metadata()
+    df = ported_metadata().metadata()
+    assert_frame_equal(expected_df, df)
+
+def check_signal():
+    meta_df = covidcast.metadata()
+    startdate = datetime(year=2022, month=2, day=1)
+    data_filter = (meta_df["max_time"] >= startdate)
+    signal_df = meta_df[data_filter].groupby("data_source")["signal"].agg(['unique'])
+    enddate = startdate + timedelta(days=3)
```

Comment:
Can we make this a longer window? Like 30 days? I want to make sure we're actually getting some data and not empty dataframes downstream.